security/s2a/internal/crypter/expander.go

```diff
@@ -24,14 +24,15 @@ func newDefaultHKDFExpander() hkdfExpander {
 	return &defaultHKDFExpander{}
 }
 
-func (*defaultHKDFExpander) expand(hash func() hash.Hash, secret, label []byte, length int) ([]byte, error) {
+func (*defaultHKDFExpander) expand(h func() hash.Hash, secret, label []byte) ([]byte, error) {
+	length := h().Size()
 	outBuf := make([]byte, length)
-	n, err := hkdf.Expand(hash, secret, label).Read(outBuf)
+	n, err := hkdf.Expand(h, secret, label).Read(outBuf)
 	if err != nil {
 		return nil, fmt.Errorf("hkdf.expand.Read failed with error: %v", err)
 	}
 	if n != length {
```

Comment:
`Read` is guaranteed not to return `n > length`. Should we change this check to `n < length` explicitly?

Reply:
It's fine if you want to keep it as is, but please add a comment explaining my comment here, if it is correct.

Reply:
You're correct. I think it's better to check `n < length` explicitly. Done.